**Student's t-test**

1- One-sample t-test\
2- Two-sample t-test\
    i- un-paired or indepenent t-test\
    ii- paired or relational/ dependent t-test

**One sample t-test**

Test a sample with a non standard value.

**Assumptions**\
    1- Observation in sample is independent and identically distributed.\
    2- Observation in sample is normally distributed.

**Interpretation**\
    **H0:** mean of sample is equal to the known value.\
    **H1:** mean of sample is unequal to the known value.

**Import libraries**

In [3]:
import pandas as pd
import seaborn as sns
from scipy.stats import ttest_1samp

Load Dataset

In [4]:
df = sns.load_dataset('titanic')
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [6]:
df1= df[['age', 'sex']]
df1.head()

age     sex
0  22.0    male
1  38.0  female
2  26.0  female
3  35.0  female
4  35.0    male

In [12]:
df1= df[['age', 'sex','fare']]
df1.head()

age     sex     fare
0  22.0    male   7.2500
1  38.0  female  71.2833
2  26.0  female   7.9250
3  35.0  female  53.1000
4  35.0    male   8.0500

Data description

In [13]:
df1.describe()

age        fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

Note: std shows data is not normal

Check the age and compare with a known value of 45 years

In [14]:
ttest_1samp(df1['age'], 45)

Ttest_1sampResult(statistic=nan, pvalue=nan)

In [15]:
# Compare with 50 years age
ttest_1samp(df1['age'], 50)
stat, p= ttest_1samp(df1['age'], 50)
print('stat= %.3f, p %.3f' % (stat, p))

# Adding conditional argument for ease
if p > 0.05:
    print('Probably the same distribution')
else:
    print('Probably different distribution')


stat= nan, p nan
Probably different distribution


In [16]:
ttest_1samp(df1['fare'], 50)
stat, p= ttest_1samp(df1['fare'], 50)
print('stat= %.3f, p %.3f' % (stat, p))

# Adding conditional argument for ease
if p > 0.05:
    print('Probably the same distribution')
else:
    print('Probably different distribution')

stat= -10.689, p 0.000
Probably different distribution


**Note:** Std is more than mean in the fare column, shows a flatten kurtosis.

**Two sample t-test**

i- **Independent student's t-test**

**Assumptions**\
    1- Observations in sample is independent and identically distributed.\
    2- Observations in sample is normally distributed.\
    3- Observations in sample have same variance.

**Interpretation**\
    **H0:** mean of samples are equal.\
    **H1:** mean of samples areunequal.

In [18]:
#Now we will compare age of male and female

# 1- Split the dataset
df_male = df1.loc[df1['sex']== 'male']
df_female = df1.loc[df1['sex']== 'female']

Import Libraries

In [20]:
from scipy.stats import ttest_ind

stat, p= ttest_ind(df_male['age'], df_female['age'])

print('stat= %.3f, p %.3f' % (stat, p))

# Adding conditional argument for ease
if p > 0.05:
    print('Probably the same distribution')
else:
    print('Probably different distribution')

stat= nan, p nan
Probably different distribution


In [21]:
df_male.describe()

age        fare
count  453.000000  577.000000
mean    30.726645   25.523893
std     14.678201   43.138263
min      0.420000    0.000000
25%     21.000000    7.895800
50%     29.000000   10.500000
75%     39.000000   26.550000
max     80.000000  512.329200

In [22]:
df_female.describe()

age        fare
count  261.000000  314.000000
mean    27.915709   44.479818
std     14.110146   57.997698
min      0.750000    6.750000
25%     18.000000   12.071875
50%     27.000000   23.000000
75%     37.000000   55.000000
max     63.000000  512.329200

**Note:** Range is too high so both can not be compared

ii- **Dependent/ Paired student's t-test**

**Assumptions**\
    1- Observations in sample is independent and identically distributed.\
    2- Observations in sample is normally distributed.\
    3- Observations in sample have same variance.\
    4- Observations across each samples are paired.

**Interpretation**\
    **H0:** mean of samples are equal.\
    **H1:** mean of samples are unequal.

In [23]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [32]:
# select male's data
df_male = df.loc[df['sex']== 'male']
df_male.head()

survived  pclass   sex   age  sibsp  parch     fare embarked  class    who  \
0         0       3  male  22.0      1      0   7.2500        S  Third    man   
4         0       3  male  35.0      0      0   8.0500        S  Third    man   
5         0       3  male   NaN      0      0   8.4583        Q  Third    man   
6         0       1  male  54.0      0      0  51.8625        S  First    man   
7         0       3  male   2.0      3      1  21.0750        S  Third  child   

   adult_male deck  embark_town alive  alone  
0        True  NaN  Southampton    no  False  
4        True  NaN  Southampton    no   True  
5        True  NaN   Queenstown    no   True  
6        True    E  Southampton    no   True  
7       False  NaN  Southampton    no  False

In [35]:
# Select only two classes
df_male_first = df_male.loc[df_male['class']== 'First']
df_male_second = df_male.loc[df_male['class']== 'Second']
df_male_third = df_male.loc[df_male['class']== 'Third']

Check the data

In [36]:
df_male_first.head()

survived  pclass   sex   age  sibsp  parch      fare embarked  class  who  \
6          0       1  male  54.0      0      0   51.8625        S  First  man   
23         1       1  male  28.0      0      0   35.5000        S  First  man   
27         0       1  male  19.0      3      2  263.0000        S  First  man   
30         0       1  male  40.0      0      0   27.7208        C  First  man   
34         0       1  male  28.0      1      0   82.1708        C  First  man   

    adult_male deck  embark_town alive  alone  
6         True    E  Southampton    no   True  
23        True    A  Southampton   yes   True  
27        True    C  Southampton    no  False  
30        True  NaN    Cherbourg    no   True  
34        True  NaN    Cherbourg    no  False

In [37]:
df_male_second.head()

survived  pclass   sex   age  sibsp  parch  fare embarked   class  who  \
17         1       2  male   NaN      0      0  13.0        S  Second  man   
20         0       2  male  35.0      0      0  26.0        S  Second  man   
21         1       2  male  34.0      0      0  13.0        S  Second  man   
33         0       2  male  66.0      0      0  10.5        S  Second  man   
70         0       2  male  32.0      0      0  10.5        S  Second  man   

    adult_male deck  embark_town alive  alone  
17        True  NaN  Southampton   yes   True  
20        True  NaN  Southampton    no   True  
21        True    D  Southampton   yes   True  
33        True  NaN  Southampton    no   True  
70        True  NaN  Southampton    no   True

In [38]:
df_male_third.head()

survived  pclass   sex   age  sibsp  parch     fare embarked  class  \
0          0       3  male  22.0      1      0   7.2500        S  Third   
4          0       3  male  35.0      0      0   8.0500        S  Third   
5          0       3  male   NaN      0      0   8.4583        Q  Third   
7          0       3  male   2.0      3      1  21.0750        S  Third   
12         0       3  male  20.0      0      0   8.0500        S  Third   

      who  adult_male deck  embark_town alive  alone  
0     man        True  NaN  Southampton    no  False  
4     man        True  NaN  Southampton    no   True  
5     man        True  NaN   Queenstown    no   True  
7   child       False  NaN  Southampton    no  False  
12    man        True  NaN  Southampton    no   True

Import Libraries

In [39]:
df_male_first.describe()

survived  pclass         age       sibsp       parch        fare
count  122.000000   122.0  101.000000  122.000000  122.000000  122.000000
mean     0.368852     1.0   41.281386    0.311475    0.278689   67.226127
std      0.484484     0.0   15.139570    0.546695    0.658853   77.548021
min      0.000000     1.0    0.920000    0.000000    0.000000    0.000000
25%      0.000000     1.0   30.000000    0.000000    0.000000   27.728100
50%      0.000000     1.0   40.000000    0.000000    0.000000   41.262500
75%      1.000000     1.0   51.000000    1.000000    0.000000   78.459375
max      1.000000     1.0   80.000000    3.000000    4.000000  512.329200

In [40]:
df_male_third.describe()

survived  pclass         age       sibsp       parch        fare
count  347.000000   347.0  253.000000  347.000000  347.000000  347.000000
mean     0.135447     3.0   26.507589    0.498559    0.224784   12.661633
std      0.342694     0.0   12.159514    1.288846    0.623404   11.681696
min      0.000000     3.0    0.420000    0.000000    0.000000    0.000000
25%      0.000000     3.0   20.000000    0.000000    0.000000    7.750000
50%      0.000000     3.0   25.000000    0.000000    0.000000    7.925000
75%      0.000000     3.0   33.000000    0.000000    0.000000   10.008300
max      1.000000     3.0   74.000000    8.000000    5.000000   69.550000

**Note:** Instances of both should be same (122 is not equal to 347)

To get a similar value:

In [41]:
df_1st = df_male_first.sample(n=100)
df_2nd = df_male_second.sample(n=100)

print("The number of instance in 1st class are=", df_1st.shape)
print("The number of instance in 2nd class are=", df_2nd.shape)

The number of instance in 1st class are= (100, 15)
The number of instance in 2nd class are= (100, 15)


In [42]:
from scipy.stats import ttest_rel
# Apply test relational to compare age of class-1 and class-3
ttest_rel(df_1st['age'], df_2nd['age'])

Ttest_relResult(statistic=nan, pvalue=nan)

In [43]:
stat, p= ttest_rel(df_1st['age'], df_2nd['age'])